java/KinesisConnectors/src/main/java/com/amazonaws/services/msf/StreamingJob.java

```diff
@@ -44,39 +48,54 @@ private static Map<String, Properties> loadApplicationProperties(StreamExecution
         }
     }
 
-    private static KinesisStreamsSource<String> createKinesisSource(Properties inputProperties) {
+    private static <T> KinesisStreamsSource<T> createKinesisSource(Properties inputProperties, final DeserializationSchema<T> deserializationSchema) {
         final String inputStreamArn = inputProperties.getProperty("stream.arn");
-        return KinesisStreamsSource.<String>builder()
+        return KinesisStreamsSource.<T>builder()
                 .setStreamArn(inputStreamArn)
                 .setSourceConfig(Configuration.fromMap(Maps.fromProperties(inputProperties)))
-                .setDeserializationSchema(new SimpleStringSchema())
+                .setDeserializationSchema(deserializationSchema)
                 .build();
     }
 
-    private static KinesisStreamsSink<String> createKinesisSink(Properties outputProperties) {
+    private static <T> KinesisStreamsSink<T> createKinesisSink(Properties outputProperties, final SerializationSchema<T> serializationSchema) {
         final String outputStreamArn = outputProperties.getProperty("stream.arn");
-        return KinesisStreamsSink.<String>builder()
+        return KinesisStreamsSink.<T>builder()
                 .setStreamArn(outputStreamArn)
                 .setKinesisClientProperties(outputProperties)
-                .setSerializationSchema(new SimpleStringSchema())
+                .setSerializationSchema(serializationSchema)
                 .setPartitionKeyGenerator(element -> String.valueOf(element.hashCode()))
                 .build();
     }
 
+    private static <T> DataStream<T> createDataStream(final StreamExecutionEnvironment env, TypeInformation<T> typeInformation, KinesisStreamsSource<T> source) {
+        return env.fromSource(source,
+                WatermarkStrategy.noWatermarks(),
+                "Kinesis source",
+                typeInformation);
+    }
+
     public static void main(String[] args) throws Exception {
         // set up the streaming execution environment
         final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         final Map<String, Properties> applicationProperties = loadApplicationProperties(env);
         LOG.warn("Application properties: {}", applicationProperties);
 
-        KinesisStreamsSource<String> source = createKinesisSource(applicationProperties.get("InputStream0"));
-        DataStream<String> input = env.fromSource(source,
-                WatermarkStrategy.noWatermarks(),
-                "Kinesis source",
-                TypeInformation.of(String.class));
+        if (applicationProperties.containsKey("SerializationType") && applicationProperties.get("SerializationType").getProperty("type").equals("JSON")) {
```

Comment:
Nobody will ever make the serialization type configurable at runtime in the real world.
And this makes the example confusibng.

Let's show the code for JSON only
[Flink docs](https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/connectors/datastream/kinesis/#kinesis-streams-source) already show how to handle a string